libs/polysemy-wire-zoo/test/Test/IntersperseSpec.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE NumDecimals #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Test.IntersperseSpec where
+
+import qualified Data.Set as S
+import Imports hiding (intersperse)
+import Polysemy
+import Polysemy.Output (output, runOutputList)
+import Polysemy.State (evalState, get, modify)
+import Polysemy.Testing
+import Polysemy.Trace
+import Test.Hspec
+import UnliftIO (async)
+
+spec :: Spec
+spec = do
+  -- This test spins up an async thread that communicates with the main
+  -- polysemy monad via an 'MVar'. We then use 'intersperse' to inject polling
+  -- logic between each bind in order to read from the 'MVar'.
+  it "should poll from async-written channel" $ do
+    result <- liftIO test
+    let desired =
+          S.fromList $
+            mconcat
+              [ fmap ("loaded: " <>) ["hello", "world", "last"],
+                fmap (show @Int) [1 .. 4],
+                ["finished"]
+              ]
+    result `shouldBe` desired
+
+  -- Example showing how intersperse lays out actions
+  it "should stick code before every action" $ do
+    let result =
+          fst $
+            run $
+              runTraceList $
+                outputToTrace show $
+                  evalState @Int 0 $
+                    intersperse ((output =<< get) <* modify (+ 1)) $ do
+                      -- 0
+                      trace "start"
+                      pure ()
```

Comment:
I think `hlint` will complain about this if we don't add an ignore pragma. 👍

Reply:
Thanks!